CONTRIBUTING.md

```diff
@@ -1,98 +1,95 @@
+If you want to implement a tool from scratch, surely read the [**Creating a Tool**](#creating-a-tool) section for orientation on where to start from.
 
 ## Table of contents
-
-[Code of Conduct](#code-of-conduct)
-
-[How Can I Contribute?](#how-can-i-contribute)
-
-- [Reporting Bugs](#reporting-bugs)
-- [Suggesting Enhancements](#suggesting-enhancements)
+- [Code of Conduct](#code-of-conduct)
+- [Reporting Bugs Or Suggesting Enhancements](#reporting-bugs-or-suggesting-enhancements)
 - [Pull Requests](#pull-requests)
-
-[Styleguides](#styleguides)
-
+- [Creating a Tool](#creating-a-tool)
+- [Rust Style Guide](#rust-style-guide)
 - [Git Commit Messages](#git-commit-messages)
-- [Tools Styleguide](#tools-styleguide)
-- [Rust Styleguide](#rust-styleguide)
-- [Documentation Styleguide](#documentation-styleguide)
-
-[References](#references)
-
-- [Man Pages (online)](#man-pages-online)
+- [Useful References](#useful-references)
+  - [Man Pages Online](#man-pages-online)
+  - [Rust Current Version Status On Tier 3](#rust-current-version-status-on-tier-3)
 
 ## Code of Conduct
-
 The Code of Conduct for this repository [here](./CODE_OF_CONFUCT.md)
 
-## How Can I Contribute?
-
-### Reporting Bugs
+## Reporting Bugs Or Suggesting Enhancements
+In the [Issues tab](https://github.com/GrayJack/coreutils/issues), click `New issue`, select `Bug report` or `Feature request`, then add the required information and submit it.
 
-In the Issues tracker, click in the button `New issue`, then select the `Bug Report` option. After that, add the required information about the bug in the template, and submit.
+## Pull Requests
+Always direct the pull Request to the `dev` branch, try to be as descriptive as possible.
 
-### Suggesting Enhancements
-
-In the Issues tracker, click in the button `New issue`, then select the `Feature request` option. After that, add the required information about the bug in the template, and submit.
-
-### Pull Requests
-
-After forking and making your changes, always make a Pull Request to the `master` branch with a proper message saying what is changed.
+## Creating a Tool
+To start a tool, copy the template:
+```sh
+cp .template -r <TOOL_NAME>
+```
 
-## Styleguides
+Now you can start editing files inside of the new directory, here's a list.
+- Replace _"template"_ occurrences in [`build.rs`](.template/build.rs) by the TOOL_NAME.
+- Make changes in [`Cargo.toml`](.template/Cargo.toml) indicated fields.
+- Configure argument parsing with `clap` at [`cli.rs`](.template/src/cli.rs), here are some rules:
+  - Every `Arg` needs `.help()`, `.long()` and `.short()` options.
+  - Short options respect the [POSIX standard](https://pubs.opengroup.org/onlinepubs/9699919799/idx/utilities.html).
+  - Long options need to match the widely common GNU's names.
```

Comment:
What about if the feature are a extension? It's not defined in the standard, then what should the person implementing the tool suppose to do?

I think it should look in the extensions of the other coreutils implementation manual pages and see if what they are doing were done before and use that for `short` and `long`, and if only one of the two is defined, then feel free to create a new one.
If it's something new, feel free to create a `short` and a `long`.

Long options should prefer match what is called on the majority of the other implementation of the tools, and if makes sense to change, add a visible alias with the common long name.
It's also nice to have alias for other long names, if there is more than one, for compatibility reasons.

Reply:
I rewrote some, commiting.